ksqldb-execution/src/main/java/io/confluent/ksql/schema/query/QuerySchemas.java

```diff
@@ -118,47 +133,83 @@ public synchronized void trackSerdeOp(
     return Collections.unmodifiableMap(loggerToSchemas);
   }
 
-  /**
-   * Builds complete SchemaInfo for a topic:
-   */
-  public SchemaInfo getTopicInfo(final String topicName) {
+  Map<String, Map<Boolean, Set<String>>> getTopicsToLoggers() {
+    return Collections.unmodifiableMap(topicsToLoggers);
+  }
+
+  public TopicFormatsInfo getTopicFormatsInfo(final String topicName) {
```

Comment:
The only  =usage of `getTopicInfo()` when called was to obtain key/value formats. I renamed to reflect the right usage. Also, in case two loggers with value formats are detected, it is simpler to combine them if the loggers use the same key/value format. With the old method, the formats were the same but the schema was different.

Reply:
I don't love that we've added a new class that's pretty much identical to an existing one. The only difference between the existing SchemaInfo and the new TopicFormatsInfo is that there's no schema and the formats are no longer optional. If not for the latter change, SchemaInfo could extend TopicFormatsInfo which would be less duplication of code. 

The current merge() method for SchemaInfo doesn't actually verify that the schemas are the same, which admittedly is odd but I think it also means it's fine if the different loggers have different schemas? What's the problem with using the existing SchemaInfo rather than introducing a new class?

Reply:
I did another change here. i didn't like the new class either. I created a `MultiSchemaInfo` now which it just contains a list of SchemaInfo for keys and another for values. This `getTopicInfo` will not throw exceptions if the schemas and/or formats are different.